hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -44,6 +45,23 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
 		return err
 	}
 
+	// Set MaxUnavailable for the inplace upgrader to use
+	intOrPercent := intstr.FromInt(1)
+	if nodePool.Spec.Management.InPlace != nil {
+		if nodePool.Spec.Management.InPlace.MaxUnavailable != nil {
+			intOrPercent = *nodePool.Spec.Management.InPlace.MaxUnavailable
+		}
+	}
+	replicas := int(k8sutilspointer.Int32PtrDerefOr(nodePool.Spec.Replicas, 0))
```

Comment:
should this check nodePool.status.Replicas?

Reply:
Hmm, the way I think about it is, you are setting a maximum unavailable for the amount of nodes in the pool you expect to be there.

If I understand it correct then, the tradeoff here is that:
1. if we use status, during spin up (as status.replicas go up), we would have multiple updates to the MaxUnavailable, which may not be ideal compared to jumping directly to the final value
2. if we use spec, something may be causing replicas to not come up, which may cause a problem depending on what stage they got to and how many actual nodes are running at the time of inplace upgrading

I don't know if 2 would cause an issue but it does sound like 1 is potentially safer... then again, if spec!=status for a long period of time, I think the cluster would have bigger issues.

Did I capture the tradeoff correctly there? I don't mind switching but just wanted to make sure I understand why

Reply:
yes thanks, let use spec.replicas